### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by Kimutai01**
#### Design
- You should follow these [design guidelines](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), including:
  - Colors.
  - Typography: font face, size and weight.
  - Layout: composition and space between elements.

> NOTE: In these design guidelines there are several UIs that you won't need for this exercise; also, some pages are not given a design and you will create them following the design guidelines of the other pages.

Original design idea by [Gregoire Vella on Behance](https://www.behance.net/gregoirevella).

The [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/) requires that you give appropriate credit to the author. Therefore, you must do it in the README of your project.

#### Interactions
- Splash screen
  - A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.

- Sign up and log in pages
  - The user should be able to register in the app with full name, email and password (all mandatory).
  - The user can log into the app using email and password.
  - If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

- Home page (catagories page)
  - When the user logs in, they are presented with the categories page.
  - For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
  - When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
  - There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

- Transactions page
  - For a given category, the list of transactions is presented, ordered by the most recent.
  - At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
  - There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new category" page
  - The user fills out a form to create a new category, indicating their name and icon (both mandatory).
  - The user click (or taps) the "Save" button to create the new category, and is taken to the home page on success.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new transaction" page
  - The user fills out a form to create a new transaction with:
    - name (mandatory)
    - amount (mandatory)
    - categories (mandatory at least one)
  - The user click (or taps) the "Save" button to create the new transation, and is taken to the transactions page for that category.
  - When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

#### Testing requirements
- Create unit and integration tests for all the most important components of your RoR application.

### CODE REVIEW

File: app/views/devise/sessions/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    <%= f.password_field :password, placeholder: "Password", class: "w-full border border-[#E9EDF4] py-3 px-5 text-base placeholder-[#ACB6BE] outline-none focus-visible:shadow-none focus:border-primary" %>
+  </div>
+
+<% end %>
+
+<div class="flex flex-col items-center w-full gap-4 my-2 text-base">
+<span class="hover:underline hover:text-primary"><%= link_to "Forgot your password?", new_password_path(resource_name) %></span>
+
+<p> Not a member yet? <span class="text-primary hover:underline"> <%= link_to "Sign up", new_registration_path(resource_name) %></span>
+</p>
```
Comment (ShoiraTa) [inline app/views/devise/sessions/new.html.erb:21]: - [x] The deployed page has error  when I click on `sign up` or `log in` links. Please make sure that the project is deployed properly, the home and detail page works as expected.
Screenshot 2022-10-12 at 22 04 21

File: README.md [hunk 2]
```diff
@@ -1,24 +1,68 @@
...
+# Budgit - A finance tracker app
 
-This README would normally document whatever steps are necessary to get the
-application up and running.
+A web application that enables you to keep track of budget
 
-Things you may want to cover:
 
-* Ruby version
+### Live demo: [budget app](https://floating-ocean-44495.herokuapp.com/)
```
Comment (ShoiraTa) [inline README.md]: - [x] Please add a link to video presentation as well 👍🏻

File: config/storage.yml [hunk 3]
```diff
@@ -7,13 +7,19 @@ local:
...
+  secret_access_key: <%= Figaro.env.spaces_secret_access_key %>
+  region: nyc3
+  bucket: budgit-app
+  endpoint: https://nyc3.digitaloceanspaces.com/
 
+cloudinary:
+  service: Cloudinary
+  cloud_name: deyemiobaa
+  api_key: <%= Figaro.env.cloud_api_key %>
+  api_secret: <%= Figaro.env.cloud_api_secret %>
```
Comment (ShoiraTa) [inline config/storage.yml:22]: - [ ] Kudos on using cloudinary 👏🏻

File: app/views/layouts/application.html.erb [hunk 4]
```diff
@@ -1,7 +1,7 @@
 <!DOCTYPE html>
 <html>
   <head>
```
Comment (ShoiraTa) [inline app/views/layouts/application.html.erb:3]: - [ ] When I start the project locally I'm getting an error:
Screenshot 2022-10-12 at 22 19 38

to recreate the issue, you can clone the repository, and try to start it.

**Review by ShoiraTa — CHANGES_REQUESTED** [review]
Hi @Kimutai01 ,

:thumbsup: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.

**First let me to point out:**

- Great commit history! ✔️ 
- You passed all the Linters! ✔️ 
- Professional pull request! ✔️ 

Keep on coding, good job so far! :clap:

There are some issues that you still need to work on to go to the next project but you are almost there!



 ## Required Changes  :recycle: 

_Check the comments under the review._ 

***
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear by mentioning my Github username [@shoirata].


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

* * *
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/layouts/application.html.erb (hunk 4 above)
Comment (Kimutai01) [reply app/views/layouts/application.html.erb:3]: Hello @ShoiraTa I used tailwind so you have to yarn install and yarn build:css

File: app/views/devise/sessions/new.html.erb (hunk 1 above)
Comment (Kimutai01) [reply app/views/devise/sessions/new.html.erb:21]: The deployment has a problem and its not working but locally it is working

File: app/views/devise/sessions/new.html.erb [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+  <div class="field">
+    <%= f.label :password, class: "sr-only" %>
+    <%= f.password_field :password, placeholder: "Password", class: "w-full border border-[#E9EDF4] py-3 px-5 text-base placeholder-[#ACB6BE] outline-none focus-visible:shadow-none focus:border-primary" %>
+  </div>
+
+<% end %>
+
+<div class="flex flex-col items-center w-full gap-4 my-2 text-base">
+<span class="hover:underline hover:text-primary"><%= link_to "Forgot your password?", new_password_path(resource_name) %></span>
```
Comment (alexander16108) [inline app/views/devise/sessions/new.html.erb:18]: current as the previous reviewer mentioned the deployed page is not working and can be viewed on this page 

image

**Review by alexander16108 — CHANGES_REQUESTED** [review]
Hi @Kimutai01 , 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!





## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: app/views/devise/sessions/new.html.erb (hunk 5 above)
Comment (Kimutai01) [reply app/views/devise/sessions/new.html.erb:18]: I have updated the link to [LIve Link](https://peaceful-river-33336.herokuapp.com/)

**Review by OLIPLICHE — APPROVED** [review]
Hi @Kimutai01, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Highlights
- The App is well deployed👏 👏
- 
Capture


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**